File: src/main/java/ua/softgroup/matrix/server/service/WorkTimePeriodService.java
package ua.softgroup.matrix.server.service;

import ua.softgroup.matrix.server.persistent.entity.WorkDay;
import ua.softgroup.matrix.server.persistent.entity.WorkTimePeriod;

import java.util.Optional;

public interface WorkTimePeriodService extends GeneralEntityService<WorkTimePeriod> {

    Optional<WorkTimePeriod> getLatestPeriodOf(WorkDay workDay);

}

File: src/main/scala/ua/softgroup/matrix/server/service/WorkTimePeriodService.scala
package ua.softgroup.matrix.server.service

import java.util.Optional

import ua.softgroup.matrix.server.persistent.entity.{WorkDay, WorkTimePeriod}

/**
  * @author Oleksandr Tyshkovets <sg.olexander@gmail.com>
  */
trait WorkTimePeriodService extends GeneralEntityService[WorkTimePeriod] {

  def getLatestPeriodOf(workDay: WorkDay): Optional[WorkTimePeriod]

}
